### PR TITLE
Integration test for question hint visibility

### PR DESCRIPTION
**Pull request by OrangeAndGreen**
## Summary
https://dimagi-dev.atlassian.net/browse/SAAS-13424

Added an integration test verifying hint visibility in the presence or lack of a hint.
The new test required changing a test CCZ (languages.ccz) so a form contained two questions, one with hint and one without.
After adding a second question to the form, two other tests needed to be repaired to support the change.

## Feature Flag
(none)

## Product Description
No user-facing effects, changes are to internal integration tests only.

## Safety Assurance

- [ ] If the PR is high risk, "High Risk" label is set
- [x ] I have confidence that this PR will not introduce a regression for the reasons below
- [ ] Do we need to enhance manual QA test coverage ? If yes, "QA Note" label is set correctly

### Automated test coverage

New test: FormEntryTest.testQuestionHintVisible
Also affected in same file: testIncompleteFormCreation, testSync

### Safety story

Ran tests related to languages.ccz to make sure no tests were broken by the change.

### CODE REVIEW

File: app/instrumentation-tests/src/org/commcare/androidTests/FormEntryTest.kt [hunk 1]
```diff
@@ -265,6 +277,31 @@ class FormEntryTest: BaseTest() {
...
+    @Test
+    @BrowserstackTests
+    fun testQuestionHintVisible() {
+        InstrumentationUtility.login("test_user_7", "123")
+        // Open first form
+        InstrumentationUtility.openForm(0, 0)
+        closeSoftKeyboard()
+
+        //Confirm the hint is visible
+        onView(withClassName(endsWith("ShrinkingTextView")))
```
Comment (shubham1g5) [inline app/instrumentation-tests/src/org/commcare/androidTests/FormEntryTest.kt:289]: For the sake of completeness, is it possible to also check for the correct text itself.

**Review by shubham1g5 — COMMENTED** [review]
looks great, though left a suggestion for you to explore.

File: app/instrumentation-tests/src/org/commcare/androidTests/FormEntryTest.kt (hunk 1 above)
Comment (OrangeAndGreen) [reply app/instrumentation-tests/src/org/commcare/androidTests/FormEntryTest.kt:289]: I added the check to verify that the hint text is correct.

**Review by shubham1g5 — APPROVED** [review]

**Comment by shubham1g5** [comment]
@OrangeAndGreen can we update this with `master`. The [PR build](https://jenkins.dimagi.com/job/commcare-android-pr-job-builder/463095/) is failing currently because it builds with Java 17.

**Pull request merged**